cpp/src/io/json/read_json.cu

```diff
@@ -49,17 +52,20 @@ size_t sources_size(host_span<std::unique_ptr<datasource>> const sources,
 /**
  * @brief Read from array of data sources into RMM buffer
  *
+ * @param buffer Device span buffer to which data is read
  * @param sources Array of data sources
  * @param compression Compression format of source
  * @param range_offset Number of bytes to skip from source start
- * @param range_size Number of bytes to read from source
+ * @param remaining_bytes_to_read Number of bytes to read from source
  * @param stream CUDA stream used for device memory operations and kernel launches
+ * @returns A subspan of the input device span containing data read
  */
-rmm::device_uvector<char> ingest_raw_input(host_span<std::unique_ptr<datasource>> sources,
-                                           compression_type compression,
-                                           size_t range_offset,
-                                           size_t range_size,
-                                           rmm::cuda_stream_view stream)
+device_span<char> ingest_raw_input(device_span<char> buffer,
+                                   host_span<std::unique_ptr<datasource>> sources,
+                                   compression_type compression,
+                                   size_t range_offset,
+                                   size_t remaining_bytes_to_read,
```

Comment:
looks like a small miscommunication in my previous comment. I didn't mean to rename the parameter, but to introduce a better named local variable.
```suggestion
                                   size_t range_size,
```

Reply:
Oh, my mistake. Fixed now.